control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -877,6 +909,28 @@ func (c *GatewayController) filterFiltersForExternalRefs(ctx context.Context, ro
 	return externalFilters, nil
 }
 
+func (c *GatewayController) getRelatedGatewayPolicy(ctx context.Context, gateway types.NamespacedName, resources *common.ResourceMap) (*v1alpha1.GatewayPolicy, error) {
+	var list v1alpha1.GatewayPolicyList
+
+	if err := c.Client.List(ctx, &list, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(Gatewaypolicy_GatewayIndex, gateway.String()),
+	}); err != nil {
+		return nil, err
+	}
+
+	if len(list.Items) > 0 {
+		//TODO should we error here or assume that this will happen in validation webhook
```

Comment:
this should be fine if we grab all the gateways referenced and then grab all the policies that reference the gateway and then loop through to find the one referencing the specific listener we want